Terminal.Gui/Views/TabView.cs

```diff
@@ -0,0 +1,794 @@
+using NStack;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace Terminal.Gui {
+
+	/// <summary>
+	/// A single tab in a <see cref="TabView"/>
+	/// </summary>
+	public class Tab {
+		private ustring text;
+
+		/// <summary>
+		/// The text to display in a <see cref="TabView"/>
+		/// </summary>
+		/// <value></value>
+		public ustring Text { get => text ?? "Unamed"; set => text = value; }
+
+		/// <summary>
+		/// The control to display when the tab is selected
+		/// </summary>
+		/// <value></value>
+		public View View { get; set; }
+
+		/// <summary>
+		/// Creates a new unamed tab with no controls inside
+		/// </summary>
+		public Tab ()
+		{
+
+		}
+
+		/// <summary>
+		/// Creates a new tab with the given text hosting a view
+		/// </summary>
+		/// <param name="text"></param>
+		/// <param name="view"></param>
+		public Tab (string text, View view)
+		{
+			this.Text = text;
+			this.View = view;
+		}
+	}
+
+	/// <summary>
+	/// Describes render stylistic selections of a <see cref="TabView"/>
+	/// </summary>
+	public class TabStyle {
+
+		/// <summary>
+		/// True to show the top lip of tabs.  False to directly begin with tab text during 
+		/// rendering.  When true header line occupies 3 pixels, when false only 2.
```

Comment:
We use the terms `rows` and `cols` or `columns`, not pixels. Please be consistent.